WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/CouponInputTransformer.swift

```diff
@@ -5,33 +5,24 @@ import Yosemite
 ///
 struct CouponInputTransformer {
 
-    /// Adds, deletes, or updates a coupon line input into an existing order.
-    /// Only the first coupon line is updated.
+    /// Adds a coupon line input into an existing order.
+    /// If the order already has that coupon line it does nothing
     ///
-    /// - If input is `nil`, then we remove the first existing coupon line.
-    /// - If there is no existing coupon lines, we insert the input one.
-    /// - If there are existing coupon lines replace the first coupon line
-    ///
-    static func update(input: OrderCouponLine?, on order: Order) -> Order {
-        // If input is `nil`, then we remove the first existing coupon line.
-        guard let input = input else {
-            let updatedLines = order.coupons.enumerated().map { index, line -> OrderCouponLine? in
-                if index == 0 {
-                    return nil
-                }
-                return OrderFactory.newOrderCouponLine(code: line.code)
-            }.compactMap({ $0 })
-            return order.copy(coupons: updatedLines)
+    static func append(input: String, on order: Order) -> Order {
+        guard !order.coupons.map({ $0.code }).contains(input) else {
+            return order
```

Comment:
This is a personal thing (so feel free to ignore it!) but I find the combination of `guard` and `!` very hard to read. Just for my understanding, in this block we're doing the same as the following, right?

```
let couponAlreadyExistsInOrder = order.coupons.map({ $0.code }).contains(input)
if couponAlreadyExistsInOrder { return order }
```

Reply:
Done in 0908e1a9d8e6489a7b86a0aa3795af62babc0fed I split it into another constant but left the guard with ! which in my opinion is ideal for these cases: only do something if it's already contained :)